PoS space review (#501)
0xPolygon/polygon-docs: docs/pos/how-to/validator/prerequisites.md

comments: true
---

> ## Before setting up the calidator node

Reviewer: ```suggestion
## Before setting up the validator node
```

!!! warning "Caution"
    
    We highly recommend running the validator node with a sentry node. This helps mitigate security concerns and potential issues that may arise with your validator node.

### Downloading the snapshot

It is recommended that you keep your snapshots handy before setting up the validator node. Link to download and extract the snapshots: https://snapshots.polygon.technology/

### RPC Endpoint for node setup

Validator nodes require an Ethereum-based RPC endpoint. You may use your own Ethereum node or utilize external infrastructure providers in our list. In order to see the list, click on the dropdown menu labeled **Infra Providers** in the navigation bar.

### Open necessary ports

> - Port `26656`: Heimdall service will connect your node to another node's Heimdall service using this port.

Reviewer: ```suggestion
- Port `26656`: Heimdall service connects your node to another node's Heimdall service using this port.
```

> - Port `30303`: Bor service will connect your node to another node's Bor service using this port.

Reviewer: ```suggestion
- Port `30303`: Bor service connects your node to another node's Bor service using this port.
```

See the doc on [validator best practices](../operate-validator-node/validator-best-practices.md) for tips on how to keep your validator node secure and operational.

## Mandatory checklist for validator setup

1. Please follow the below checklist in order to set up your validator node using binaries, Ansible, or packages.

    | Checklist               | Binaries                                       | Ansible                                                                 | Packages                                       |
    | ----------------------- | ---------------------------------------------- | ----------------------------------------------------------------------- | ---------------------------------------------- |

>     | Machines Required   | 2 Machines - *Sentry* & *Validator*        | 3 Machines - *Local Machine*, *Sentry* and *Validator*            | 2 Machines - *Sentry* & *Validator*        |
>     | Install Go Packages | Yes                                            | No                                                                      | No                                             |
>     | Install Python      | No                                             | Yes (only on the *Local Machine* where the *Ansible Playbook* runs) | No                                             |
>     | Install Ansible     | No                                             | Yes (only on one machine)                                               | No                                             |
>     | Install Bash        | No                                             | No                                                                      | Yes                                            |
>     | Run Build Essential | Yes                                            | No                                                                      | No                                             |
>     | Node Setup          | [Using Binaries](../validator/validator-binaries.md) | [Using Ansible](../validator/validator-ansible.md)                            | [Using Packages](../validator/validator-packages.md) |

Reviewer: ```suggestion
    | Machines required   | 2 Machines - *Sentry* & *Validator*        | 3 Machines - *Local Machine*, *Sentry* and *Validator*            | 2 Machines - *Sentry* & *Validator*        |
    | Install Go packages | Yes                                            | No                                                                      | No                                             |
    | Install Python      | No                                             | Yes (only on the *Local Machine* where the *Ansible Playbook* runs) | No                                             |
    | Install Ansible     | No                                             | Yes (only on one machine)                                               | No                                             |
    | Install Bash        | No                                             | No                                                                      | Yes                                            |
    | Run Build Essential | Yes                                            | No                                                                      | No                                             |
    | Node setup          | [Using binaries](../validator/validator-binaries.md) | [Using Ansible](../validator/validator-ansible.md)                            | [Using packages](../validator/validator-packages.md) |
```

        ```bash
        git checkout fixing_symlinks_on_clean
        git pull https://github.com/maticnetwork/node-ansible/tree/fixing_symlinks_on_
        ```

    - Cross check `clean.yml` on your machine with [`clean.yml` in the GitHub repo](https://github.com/maticnetwork/node-ansible/blob/fixing_symlinks/playbooks/clean.yml)

    - If there are any differences, replace the one on your machine with the one in the repo

    - You should be able to run the clean script now and then also be able to run the installation script

- `moniker=enter unique identifier`
    - The unique identifier asked as per the document. This can be anything, for instance your name.

- `eth_rpc_url =insert Infura or any full node RPC URL to Ethereum`
    - For this step, sign in to https://infura.io (signup if you already don't have an account)

    - Copy the `https` endpoint provided under *ethereum → keys → endpoints*

        ![Figure: Knowledge base - node setup 5](../../../img/pos/node-setup-5.png)

        - Provide the copied `https` endpoint as `eth_rpc_url` in `/var/lib/heimdall/config/heimdall-config.toml`

### 2. Why do I have to keep ETH in my signer account?

> ETH is required on your signer account because for submitting checkpoints to Ethereum, all transactions require ETH to be used as Gas. Hence, ETH is required in your signer Account.

Reviewer: ```suggestion
ETH is required on your signer account because for submitting checkpoints to Ethereum, all transactions require ETH to be used as Gas. Hence, ETH is required in your signer Account.
```
```suggestion
ETH is required on your signer account because for submitting checkpoints to Ethereum, all transactions require ETH to be used as Gas. Hence, ETH is required in your signer account.
```
